Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/remesh_planar_patches.h

```diff
@@ -0,0 +1,1649 @@
+ *      \cgalParamDefault{`boost::get(CGAL::vertex_point, tm_in)`}
+ *      \cgalParamExtra{If this parameter is omitted, an internal property map for `CGAL::vertex_point_t`
+ *                      must be available in `TriangleMeshIn`.}
+ *    \cgalParamNEnd
+ *    \cgalParamNBegin{geom_traits}
+ *      \cgalParamDescription{an instance of a geometric traits class}
+ *      \cgalParamType{a class model of `Kernel`}
+ *      \cgalParamDefault{a \cgal Kernel deduced from the point type, using `CGAL::Kernel_traits`}
+ *      \cgalParamExtra{The geometric traits class must be compatible with the vertex point type.}
+ *    \cgalParamNEnd
+ *    \cgalParamNBegin{edge_is_constrained_map}
+ *      \cgalParamDescription{a property map filled by this function and that will contain `true` if an edge is on the border of a patch and `false` otherwise.}
+ *      \cgalParamType{a class model of `ReadWritePropertyMap` with `boost::graph_traits<TriangleMeshIn>::%edge_descriptor`
+ *                     as key type and `bool` as value type}
+ *      \cgalParamDefault{None}
+ *    \cgalParamNEnd
+ *    \cgalParamNBegin{face_patch_map}
+ *      \cgalParamDescription{a property map filled by this function and that will contain for each face the id
+ *                            of its patch in the range `[0, number of patches - 1]`}
+ *      \cgalParamType{a class model of `ReadWritePropertyMap` with `boost::graph_traits<TriangleMeshIn>::%face_descriptor`
+ *                     as key type and `std::size_t` as value type}
+ *      \cgalParamDefault{None}
+ *    \cgalParamNEnd
+ *    \cgalParamNBegin{vertex_corner_map}
+ *      \cgalParamDescription{a property map filled by this function and that will contain for each vertex that is a corner
+ *                            an id in the range `[0, number of corners - 1]`, and `std::size_t(-1)` otherwise.}
+ *      \cgalParamType{a class model of `ReadWritePropertyMap` with `boost::graph_traits<TriangleMeshIn>::%vertex_descriptor`
+ *                     as key type and `std::size_t` as value type}
+ *      \cgalParamDefault{None}
+ *   \cgalParamNEnd
+ *    \cgalParamNBegin{cosine_of_maximum_angle}
+ *      \cgalParamDescription{The maximum angle, given as a cosine,
+ *                            (i) between the normals of the supporting planes of adjacent faces such that they are considered coplanar, and
+ *                            (ii) for the smallest angle between the supporting line of a segment and an adjacent segment such that they are considered collinear.}
+ *      \cgalParamType{`FT` type from the `geom_traits` parameter}
+ *      \cgalParamDefault{1, which means exact coplanarity and collinearity}
+ *   \cgalParamNEnd
+ *    \cgalParamNBegin{do_not_triangulate_faces}
+ *      \cgalParamDescription{if `true`, faces of `out` will not be triangulated, but the one with more than one connected component of the boundary.}
```

Comment:
`tm_out` (here and at another place).

Also rephrase to something like "only triangulates representing a planar region if this region is not simply connected"?